File: src/main/java/com/beat/global/external/notification/slack/vo/text/PlainText.java
package com.beat.global.external.notification.slack.vo.text;

import static com.beat.global.external.notification.slack.vo.SlackConstant.TEXT_TYPE_PLAIN;

public record PlainText(
	String type,
	String text,
	boolean emoji
) implements Text {

	public static PlainText newInstance(String text) {
		return new PlainText(TEXT_TYPE_PLAIN, text, true);
	}
}

File: infrastructure/src/main/kotlin/com/beat/infrastructure/external/notification/slack/vo/text/PlainText.kt
package com.beat.infrastructure.external.notification.slack.vo.text

import com.beat.infrastructure.external.notification.slack.vo.SlackConstant.TEXT_TYPE_PLAIN

internal data class PlainText(
    override val type: String,
    val text: String,
    val emoji: Boolean,
) : Text {
    companion object {
        fun newInstance(text: String): PlainText = PlainText(TEXT_TYPE_PLAIN, text, true)
    }
}
